Minio/V4Authenticator.cs

```diff
@@ -310,7 +312,12 @@ private string GetPresignCanonicalRequest(IRestClient client, IRestRequest reque
                 path = "/" + path;
             }
             canonicalStringList.AddLast(path);
-            canonicalStringList.AddLast(requestQuery);
+            String query = requestQuery;
+            foreach (string header in headersToSign.Keys)
+            {
+               query = query + "&" + header + "=" + utils.UrlEncode(headersToSign[header]);
+            }
```

Comment:
You are looping through dictionary, and then getting the value using the indexed acessor(that is a overhead), my suggestion does not get the overhead because it use the property value, from current dictionary entry.
```csharp
var query = headersToSign.Aggregate(requestQuery, (pv, cv) => $"{pv}&{h.Key}={utils.UrlEncode(h.Value)}");
```

Reply:
Thanks @AlbertoMonteiro, that's more concise.